lib/mix/lib/mix/tasks/compile.elixir.ex

```diff
@@ -68,6 +68,7 @@ defmodule Mix.Tasks.Compile.Elixir do
     force = opts[:force] || Mix.Utils.stale?(configs, [manifest])
 
     opts = Keyword.merge(project[:elixirc_options] || [], opts)
+    opts = opts ++ xref_exclude_to_no_warn_undefined(project)
```

Comment:
Do we need to deep merge them? Otherwise, there is a chance the options in `xref[:exclude]` will actually win over the user configuration. I think the `Keyword.update/4` will be necessary.

Reply:
My thinking was that `no_warn_undefined` could be declared multiple times, like it can with `@compile`? If that's not the case I will deep merge.